clients/slack/slack_client.py

```diff
@@ -78,15 +78,16 @@ def upload_file(self, channel_name: str, file_path: str, message: SlackMessageSc
 
     def _get_channel_id(self, channel_name: str) -> str:
         try:
+            channels = self._get_channels()
             channel = [
                 channel
                 for channel
-                in self.client.conversations_list()["channels"]
+                in channels
                 if channel["name"] == channel_name
             ][0]
             return channel["id"]
-        except Exception as e:
-            logger.error(f"Could not find the channel - {channel_name}. Error: {e}")
+        except Exception:
+            logger.error(f"Could not find the channel - {channel_name}. Elementary app's available cahnnels: {[channel.get('name') for channel in channels]}")
```

Comment:
what happens if the exception is in the get_channels?

Reply:
But still if exception that is not SlackApi is thrown from get_channels it will be caught here and channels could be undefined, maybe just initialize it here with empty list before assigning a value to it from get_channels?